app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt

```diff
@@ -126,11 +126,19 @@ constructor(
      * Data initialization variants
      */
     fun onCourseId(courseId: Long, viewSource: CourseViewSource, forceUpdate: Boolean = false) {
-        observeCourseData(courseInteractor.getCourseHeaderData(courseId), viewSource, forceUpdate)
+        val courseIdToPass = (state as? CourseView.State.CourseLoaded)
+            ?.courseHeaderData
+            ?.courseId
+            ?: courseId
```

Comment:
Course id will be the same. You have to use `courseInteractor.getCourseHeaderData(course: Course, ...)` in case of `CourseLoaded` state.